plugin/trino-opa/src/test/java/io/trino/plugin/opa/OpaAccessControlFilteringUnitTest.java

```diff
@@ -0,0 +1,362 @@
+                                        """.formatted(requestedColumnsForTable.getKey().getTableName(), column))))
+                .collect(toImmutableList());
+        assertStringRequestsEqual(expectedRequests, this.mockClient.getRequests(), "/input/action");
+        assertTrue(
+                Maps.difference(
+                        result,
+                        ImmutableMap.builder()
+                                .put(tableOne, ImmutableSet.of("table_one_column_one", "table_one_column_two"))
+                                .put(tableTwo, ImmutableSet.of("table_two_column_two"))
+                                .buildOrThrow()).areEqual());
+    }
+
+    @Test
+    public void testEmptyFilterColumns()
+    {
+        SchemaTableName someTable = SchemaTableName.schemaTableName("my_schema", "my_table");
+        Map<SchemaTableName, Set<String>> requestedColumns = ImmutableMap.of(someTable, ImmutableSet.of());
+
+        Map<SchemaTableName, Set<String>> result = authorizer.filterColumns(
+                requestingSecurityContext,
+                "my_catalog",
+                requestedColumns);
+
+        assertTrue(mockClient.getRequests().isEmpty());
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void testFilterFunctions()
+    {
+        SchemaFunctionName functionOne = new SchemaFunctionName("my_schema", "function_one");
+        SchemaFunctionName functionTwo = new SchemaFunctionName("my_schema", "function_two");
+        Set<SchemaFunctionName> requestedFunctions = ImmutableSet.of(functionOne, functionTwo);
+        this.mockClient.setHandler(buildHandler("/input/action/resource/function/functionName", "function_two"));
+
+        Set<SchemaFunctionName> result = authorizer.filterFunctions(
+                requestingSecurityContext,
+                "my_catalog",
+                requestedFunctions);
+        assertEquals(ImmutableSet.copyOf(result), ImmutableSet.of(functionTwo));
+
+        List<String> expectedRequests = requestedFunctions.stream()
+                .map(function -> """
+                    {
+                        "operation": "FilterFunctions",
+                        "resource": {
+                            "function": {
+                                "catalogName": "my_catalog",
+                                "schemaName": "%s",
+                                "functionName": "%s"
+                            }
+                        }
+                    }""".formatted(function.getSchemaName(), function.getFunctionName()))
+                .collect(toImmutableList());
+        assertStringRequestsEqual(expectedRequests, this.mockClient.getRequests(), "/input/action");
+    }
+
+    @ParameterizedTest(name = "{index}: {0}")
+    @MethodSource("io.trino.plugin.opa.FilteringTestHelpers#emptyInputTestCases")
+    public void testEmptyRequests(
+            BiFunction<OpaAccessControl, SystemSecurityContext, Collection> callable)
```

Comment:
The `Collection` type here is a raw type which disables type checking in Java.  Add something `<?>`, or even better set the actual type.